src/marqo/tensor_search/validation.py

```diff
@@ -598,3 +599,39 @@ def validate_score_modifiers_object(score_modifiers: List[dict]):
             f"Please revise your score_modifiers based on the provided error."
             f"\n Check `https://docs.marqo.ai/0.0.17/API-Reference/search/#score-modifiers` for more info."
         )
+
+
+def validate_delete_docs_request(delete_request: MqDeleteDocsRequest, max_delete_docs_count: int):
```

Comment:
I only now realise that MqDeleteDocsRequest is a NamedTuple, not a Pydantic BaseModel. I think we'd find that if we use a BaseModel, most of this is automatically validated at __init__ time.

Reply:
And I think you can do both custom, and extended validation for something like `if (len(delete_request.document_ids) > max_delete_docs_count) and max_delete_docs_count is not None:`